### ISSUE TITLE
[docs] Add library/API usage documentation (EN+JA)

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#221]
## Background / goal

The deployed docs site currently covers installation, CLI, Studio, contributor notes, and example configs — but almost none of the actual Python API (`FastDiffusionModel`, `DiffusionTrainer`, collators, schedulers, evaluators, model families) is documented. Users who want to call unturtle from code have to read the source. Expand the docs with task-oriented Library guides plus a compact API reference, in both EN and JA.

This issue depends on #220 being merged so that `starlight-links-validator` can gate the large PR's internal links at build time.

## Approach (hybrid)

- **Task-oriented guides** covering end-to-end flows (~100–250 lines each).
- **Compact API reference pages**: signature + key args + minimal example + GitHub source link (pinned to `main`). Not an exhaustive argument dump.
- **EN + JA added in the same PR.** Zero-missing-page rule across locales.

## New files (28 total: 14 EN + 14 JA)

```
docs/src/content/docs/
├── library/
│   ├── index.md                 # Library overview
│   ├── quickstart.md            # G1
│   ├── fast-diffusion-model.md  # G2
│   ├── sft.md                   # G3
│   ├── block-diffusion.md       # G4
│   ├── grpo.md                  # G5
│   ├── generation.md            # G6
│   └── evaluation.md            # G7
├── reference/
│   ├── trainers.md
│   ├── collators.md
│   ├── alpha-schedulers.md
│   ├── evaluators.md
│   └── models.md                # G8 (A2D / LLaDA / Dream / ModernBERT combined)
└── ja/...                       # mirror of all 14 pages
```

## API coverage per guide

| Guide | Public symbols covered |
|---|---|
| G1 Quickstart | `FastDiffusionModel.from_pretrained`, `DiffusionTrainer`, `DiffusionTrainingArguments` (minimal) |
| G2 FastDiffusionModel | `FastDiffusionModel.from_pretrained`, `for_inference`, LoRA, quantization |
| G3 SFT | `DiffusionTrainer`, `DiffusionTrainingArguments`, `MaskedDiffusionDataCollator`, `PackedMaskedDiffusionDataCollator`, `Linear/CosineAlphaScheduler`, `make_alpha_scheduler` |
| G4 Block diffusion | `BlockDiffusionTrainer`, `BlockDiffusionTrainingArguments`, `BlockDiffusionDataCollator`, `create_block_diffusion_attention_mask` |
| G5 Diffu-GRPO | `DiffuGRPOTrainer`, `DiffuGRPOConfig`; note on optional `trl` / `mergekit` deps |
| G6 Generation | `FastDiffusionModel.for_inference`, Dream block-decode, LLaDA denoising |
| G7 Evaluation | `BaseEvaluator`, `GenerationEvaluator`, `MaskedDiffusionEvaluator` |
| G8 Models reference | `A2D{Llama,Qwen2,Qwen3,ModernBert}*`, `LLaDA*`, `Dream*`, `DiffusionModernBert*` — config fields and per-family gotchas distilled from `CLAUDE.md`'s Model Path Reminders (do not leak `dev/repos/` internal references) |

## Sidebar changes

Add two top-level sections to `docs/astro.config.mjs`:

- **Library** (`ja: ライブラリ`): Overview, Quickstart, FastDiffusionModel, Masked diffusion SFT, Block diffusion, Diffu-GRPO, Generation, Evaluation.
- **API reference** (`ja: APIリファレンス`): Trainers, Data collators, Alpha schedulers, Evaluators, Model families.

## Page template

Every guide (G1–G7) must contain:
1. Overview (1–2 paragraphs, what you learn / prerequisites).
2. Minimal runnable code example.
3. Key options (important args and behavior).
4. Common patterns (2–3 snippets).
5. Pitfalls (pull from CLAUDE.md/AGENTS.md gotchas).
6. Related (links to API reference pages and other guides).

## Integration touch-ups

- `index.mdx` + `ja/index.mdx`: add a Library CardGrid entry and a "Library guides" bullet under Main entrypoints.
- `cli.md` + `ja/cli.md`: add a one-line pointer to `/library/` for programmatic usage.
- `getting-started/installation.md` + `ja/...`: add `library/quickstart/` to Next steps.

## EN/JA sync rules

- Mirror file structure exactly; zero-missing-page rule.
- Sidebar items use `translations: { ja: '...' }`.
- Code blocks identical character-for-character; only comments may differ.
- JA pages link to other JA pages.

## Acceptance criteria

- [ ] 28 new pages land in a single PR.
- [ ] `astro build` passes with `starlight-links-validator` enabled (from #220).
- [ ] EN and JA have identical page sets.
- [ ] Post-merge, `/library/`, `/library/quickstart/`, `/library/sft/`, `/reference/models/` and their `/ja/...` counterparts return 200.
- [ ] Dogfood: at least the G1 Quickstart snippet is manually run against the current repo to confirm it does not ImportError or break on the advertised signatures.

## Scope exclusions (YAGNI)

- Advanced Triton kernels walkthrough (G9) and standalone alpha-scheduler page (G10).
- sphinx-style autogenerated reference.
- External link checker (lychee).
- Docs versioning, `llms.txt`, benchmarks pages.

## Related

- Depends on #220.
